USAGE.md

```diff
@@ -1,370 +1,707 @@
-    ])
-    
-get_group_by_alias(fs, 'Cozy Bear')[0]
+t1134 = src.query([ 
+    stix2.Filter("external_references.external_id", "=", "T1134"), 
+    stix2.Filter("type", "=", "attack-pattern")
+])[0]
 ```
 
-### Get all Techniques used by a Group
-We query for the techniques that are directly connected to a group.  This does NOT include techniques which are only used by the group's software.
+The old 1:1 mitigations causing this issue are deprecated, so you can also filter them out that way — see [removing revoked and deprecated objects](#Removing-revoked-and-deprecated-objects), below.
+
+### By Name
+The following recipe retrieves an object according to its name:
 
 ```python
-def get_technique_by_group(src, stix_id):
-    relations = src.relationships(stix_id, 'uses', source_only=True)
-    return src.query([
+def get_technique_by_name(thesrc, name):
+    filt = [
         Filter('type', '=', 'attack-pattern'),
-        Filter('id', 'in', [r.target_ref for r in relations])
-    ])
-
-group = get_group_by_alias(fs, 'Cozy Bear')[0]
-get_technique_by_group(fs, group)
+        Filter('name', '=', name)
+    ]
+    return thesrc.query(filt)
+# get the technique titled "System Information Discovery"
+get_technique_by_name(src, 'System Information Discovery')
 ```
 
-### Get all Techniques used By a Group's Software
-This example is fairly complex.  First we must query the group's relationships to find the software that the group uses.  Then we must use the relationships for each piece of software in order to find the techniques.  Notice the expression for `software_uses` involves directly filtering the relationship objects rather than using the `relationships` method of the DataSource `src`.  This is because, as of this writing, calls to `query` are expensive, and it is better to minimize them.
+### By Alias
+The following methodology can be used to find the group corresponding to a given alias:
 
 ```python
-from stix2.utils import get_type_from_id
+def get_group_by_alias(thesrc, alias):
+    return thesrc.query([
+        Filter('type', '=', 'intrusion-set'),
+        Filter('aliases', '=', alias)
+    ])[0]
+    
+get_group_by_alias(src, 'Cozy Bear')
+```
 
-def get_techniques_by_group_software(src, group_stix_id):
-    # get the malware, tools that the group uses
-    group_uses = [
-        r for r in src.relationships(group_stix_id, 'uses', source_only=True)
-        if get_type_from_id(r.target_ref) in ['malware', 'tool']
-    ]
 
-    # get the technique stix ids that the malware, tools use
-    software_uses = src.query([
-        Filter('type', '=', 'relationship'),
-        Filter('relationship_type', '=', 'uses'),
-        Filter('source_ref', 'in', [r.source_ref for r in group_uses])
-    ])
 
-    #get the techniques themselves
-    return src.query([
-        Filter('type', '=', 'attack-pattern'),
-        Filter('id', 'in', [r.target_ref for r in software_uses])
-    ])
+## Getting multiple objects
```

Comment:
The previous section, `Getting an object`, has consistent naming for its subsections ("By ________"). It would be nice if the subsections for `Getting multiple objects` could be consistent in a similar manner (currently, some have "Getting", some don't, one just uses "Get").

Reply:
Good idea. I've tried to do this as best as possible in the new version, but sometimes it doesn't work quite as well as I'd like (e.g "Get Techniques by Tactic" still needs to say the type, so it becomes "Techniques by Tactic"

I'm therefore changing all of the subsections in this section to start with the type, e.g "Objects by type" or "Techniques by Platform."